crates/circuit/src/dag_circuit.rs

```diff
@@ -233,9 +233,9 @@ pub struct DAGCircuit {
     cregs: Py<PyDict>,
 
     /// The cache used to intern instruction qargs.
-    qargs_interner: Interner<[Qubit]>,
+    pub(crate) qargs_interner: Interner<[Qubit]>,
     /// The cache used to intern instruction cargs.
-    cargs_interner: Interner<[Clbit]>,
+    pub(crate) cargs_interner: Interner<[Clbit]>,
```

Comment:
This change is a little concerning as it would allow you to change the original interner from the received `DAGCircuit`. Wouldn't it be better to expose these via a view or immutable reference, like it's being done for #13117?

Edit: I just saw your [comment](https://github.com/Qiskit/qiskit/pull/13117#discussion_r1752880546), I'll keep this comment here as a reminder to change this oncce #13117 merges.

Reply:
Done in: https://github.com/Qiskit/qiskit/pull/13126/commits/7549c2db92ce2779dfc89b92b8e56e7fb6414b14